backend/src/data/repository/post.ts

```diff
@@ -0,0 +1,20 @@
+import { DTO } from '../../interface';
+import { Post } from '../models/post';
+
+export class PostRepository {
+  async getAllPostById(ID: DTO.ID): Promise<DTO.IPost[]> {
+    const posts = await Post.find({ userId: ID }).populate('creator', ['firstName', 'lastName', 'photo']);
+    return posts;
+  }
+
+  async createPost(newPost: DTO.IPost) {
+    const post = new Post(newPost);
+    await post.save(function (err, post) {
+      if (err) return console.error(err);
+      console.log(post);
+    });
+    return post;
+  }
```

Comment:
potential issue here - we creating post and return this directly. Not the saved copy which comes as second param from callback function